db/DB.js

```diff
@@ -264,18 +279,35 @@ const DB = {
     await knex("subscribers").where({"id": subscriber.id}).del();
   },
 
-  async removeSubscriberByEmail(email) {
-    const sha1 = getSha1(email);
-    return await this._getSha1EntryAndDo(sha1, async aEntry => {
-      await knex("subscribers")
-        .where("id", "=", aEntry.id)
+  // This is used by SES callbacks to remove email addresses when recipients
+  // perma-bounce or mark our emails as spam
+  // Removes from either subscribers or email_addresses as necessary
+  async removeEmail(email) {
+    const subscriber = await this.getSubscriberByEmail(email);
+    if (!subscriber) {
+      const emailAddress = await this.getEmailAddressByEmail(email);
+      if (!emailAddress) {
+        log.warn("removed-subscriber-not-found");
+        return;
+      }
+      await knex("email_addresses")
+        .where({
+          "email": email,
```

Comment:
I think I'm confused here.
Above on L286 we're trying to find a `subscriber` by `email`. But here we're in a `if (!subscriber) {...}` block, so I'm assuming we didn't find a match, but here on L295, we're trying to find a match `WHERE email = email`.
Also curious is that we're trying to do a `emailAddress = this.getEmailAddressByEmail(email)`, but then not really "using" the `emailAddress` after that. Not sure if this line should be `.where({"email": emailAddress, ...)`.

Reply:
The first check looks in the `subscribers` table. If we can't find the email address in there, the next check looks in the `email_addresses` table.

I'm using the `if (!emailAddress)` just to check if it wasn't found and log the warning in that case.